Content.Client/Stylesheets/Redux/Fonts/FontStack.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Content.Client.Resources;
+using JetBrains.Annotations;
+using Robust.Client.Graphics;
+using Robust.Client.ResourceManagement;
+
+namespace Content.Client.Stylesheets.Redux.Fonts;
+
+[PublicAPI]
+public abstract class FontStack(IResourceCache resCache)
+{
+    /// <summary>
+    ///     The primary font path, with string substitution markers.
+    /// </summary>
+    /// <remarks>
+    ///     If using the default GetFontPaths function, the substitutions are as follows:
+    ///     0 is the font kind.
+    ///     1 is the font kind with BoldItalic replaced with Bold when it occurs.
+    /// </remarks>
+    public abstract string FontPrimary { get; }
+
+    /// <summary>
+    ///     The symbols font path, with string substitution markers.
+    /// </summary>
+    /// <remarks>
+    ///     If using the default GetFontPaths function, the substitutions are as follows:
+    ///     0 is the font kind.
+    ///     1 is the font kind with BoldItalic replaced with Bold when it occurs.
+    /// </remarks>
+    public abstract string FontSymbols { get; }
+
+    /// <summary>
+    ///     The fallback font path, exactly. (no string substitutions.)
+    /// </summary>
+    public virtual string FontFallback => "/EngineFonts/NotoSans/NotoSans-Regular.ttf";
+
+    /// <summary>
+    ///     Any extra fonts that should be stuck in after Symbols but before the fallback.
+    /// </summary>
+    public abstract string[] Extra { get; }
+
+    public virtual HashSet<FontKind> AvailableKinds => new()
+        { FontKind.Regular, FontKind.Bold, FontKind.Italic, FontKind.BoldItalic };
+
+    /// <summary>
+    ///     This should return the paths of every font in this stack given the abstract members.
+    /// </summary>
+    /// <param name="kind">Which font kind to use.</param>
+    /// <returns>An array of </returns>
+    protected virtual string[] GetFontPaths(FontKind kind)
+    {
+        if (!AvailableKinds.Contains(kind))
+        {
+            if (kind == FontKind.BoldItalic && AvailableKinds.Contains(FontKind.Bold))
+            {
+                kind = FontKind.Bold;
+            }
+            else
+            {
+                kind = FontKind.Regular;
+            }
+        }
+
+        var simpleKindStr = kind.SimplifyCompound().AsFileName();
+        var boldOrRegularStr = kind.RegularOr(FontKind.Bold).AsFileName();
+
+        var kindStr = kind.AsFileName();
+        var fontList = new List<string>()
+        {
+            string.Format(FontPrimary, kindStr, simpleKindStr, boldOrRegularStr),
+            string.Format(FontSymbols, kindStr, simpleKindStr, boldOrRegularStr),
+        };
+        fontList.AddRange(Extra);
+        fontList.Add(FontFallback);
+        return fontList.ToArray();
+    }
```

Comment:
I think this was just pretty much copied over verbatim from `StyleNano.cs` and moony probably intended to go back and rework it at some point. 

I can alternatively rework it to specify different paths for each font face, or you can do the engine PR. Up to you.

Reply:
leaving all font-related comments unresolved for now

Reply:
If this is just a holdover from `StyleNano` then I'd say this definitely shouldn't have been moved into a "general purpose" looking thing like this. If this can be moved into `NotoFontStack` and the base type being left as an "uuhhh" for now, that's good enough IMO.